p2p/async-buffer/tests/basic.rs

```diff
@@ -0,0 +1,37 @@
+use futures::{FutureExt, StreamExt};
+
+use async_buffer::new_buffer;
+
+#[tokio::test]
+async fn async_buffer_send_rec() {
```

Comment:
IMO `loom` tests should always be present when introducing hand-written data structures involving async and/or atomics.

Could be left as a TODO for a big "introduce loom" PR.

Reply:
Less of a concern on SPSC.